### PR TITLE
Fixes #1792 NullReferenceException while typing parameter list

### LINKED ISSUE #1792: NullReferenceException while typing parameter list
Dev 15 RC

My code:
```
import unittest
#import boo

class Test_test1(unittest.TestCase):
    def test_A(self):
        self.fail("Not implemented")

    def test_B(self):
        self.assertEqual(4, 4)

if __name__ == '__main__':
    unittest.main()

```

I had just typed the second 4.

Analysis of my Anaconda environment is ongoing.

```
System.NullReferenceException: Object reference not set to an instance of an object. at 
Microsoft.PythonTools.Intellisense.IntellisenseController.UpdateCurrentParameter() at
 Microsoft.PythonTools.Intellisense.IntellisenseController.HandleChar(Char ch) at
 Microsoft.PythonTools.Intellisense.IntellisenseController.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut) at 
Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut) at 
Microsoft.VisualStudio.Editor.Implementation.CommandChainNode.InnerExec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut) at
 Microsoft.VisualStudio.Editor.Implementation.SimpleTextViewWindow.Exec(Guid& pguidCmdGroup, UInt32 nCmdID, UInt32 nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut) at
 Microsoft.VisualStudio.Editor.Implementation.VsKeyboardFilter.SendCommand(Guid cmdGroup, UInt32 cmdID, Object inParam) at
 Microsoft.VisualStudio.Editor.Implementation.VsKeyboardFilter.TextInput(TextCompositionEventArgs args) at Microsoft.VisualStudio.Text.Editor.Implementation.KeyProcessorDispatcher.<>c.<DispatchTextInputEvents>b__6_0(KeyProcessor p, TextCompositionEventArgs args) at
 Microsoft.VisualStudio.Text.Editor.Implementation.KeyProcessorDispatcher.<>c__DisplayClass10_1`1.<Dispatch>b__0() at
 Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)
```

### PR DESCRIPTION
**Pull request by zooba**
Fixes #1792 NullReferenceException while typing parameter list
Protects against null references in UpdateCurrentParameter.

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs [hunk 1]
```diff
@@ -716,67 +716,67 @@ private class ExpressionCompletionWalker : PythonWalker {
...
-                );
+            var prevBuffer = sig.ApplicableToSpan.TextBuffer;
+            var textBuffer = _textView.TextBuffer;
 
-                if (targetPt == null) {
-                    return;
-                }
-                var span = targetPt.Value.Snapshot.CreateTrackingSpan(targetPt.Value.Position, 0, SpanTrackingMode.EdgeInclusive);
+            var targetPt = _textView.BufferGraph.MapDownToFirstMatch(
+                new SnapshotPoint(_textView.TextBuffer.CurrentSnapshot, position),
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs]: did you want to use the new `textBuffer` variable here?

Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs]: All I really did was unindent the code by one level, so there's not much new here. It looks like that variable was and always has been unused. But I'll switch to using it, as it makes sense.

Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs]: Actually, I'll just remove the unused variable. It wouldn't prevent a race against the buffer changing anyway, but I think we're on the UI thread here so it doesn't matter.

**Comment by huguesv** [comment]
👍

**Pull request merged**